client/src/contexts/UserContext.js

```diff
@@ -0,0 +1,33 @@
+import { createContext, useState, useContext, useEffect } from "react";
+
+const UserContext = createContext();
+
+const UserContextProvider = (props) => {
+  const [user, setUser] = useState("Victor");
+  const [token, setToken] = useState(() => {
+    const tokenValue = localStorage.getItem("token");
+    return tokenValue;
+  });
+
+  useEffect(() => {
+    console.log("fetching user data");
```

Comment:
Let's wrap this fetching (now console.log, but later fetching data) with `if` statement that checks if the token is `true` before fetching.

Something like below:
```js
  if (token) { 
      console.log("fetching user data");
    }
```